runtime/executor/src/main/java/org/apache/nemo/runtime/executor/data/MemoryAllocationException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.nemo.runtime.executor.data;
+
+/**
+ * An exception to be thrown when a memory allocation operation is not successful.
+ */
+public class MemoryAllocationException extends Exception {
```

Comment:
I also noticed that this exception is never thrown in this PR. I suppose it will be used in the upcoming implementation, regarding the off-heap region separation (thrown when the off-heap memory allocation exceeds the region specification). I will add the description to this exception and remove the unnecessary `catch` clauses.

Reply:
I realized I missed this part. It is supposed to be thrown while allocating direct `ByteBuffer`, because it will throw an out-of-memory error if the system lacks memory to be allocated. I will add this part so that this exception is used.